src/vmm/src/guest_config/x86_64/cpuid/amd/normalize.rs

```diff
@@ -292,18 +292,19 @@ impl super::AmdCpuid {
     ) -> Result<(), ExtendedCacheTopologyError> {
         for i in 0.. {
             if let Some(subleaf) = self.get_mut(&CpuidKey::subleaf(0x8000001d, i)) {
-                // Cache type. Identifies the type of cache.
+                // Cache level. Identifies the level of this cache. Note that the enumeration value
+                // is not necessarily equal to the cache level.
                 // ```text
                 // Bits Description
-                // 00h Null; no more caches.
-                // 01h Data cache
-                // 02h Instruction cache
-                // 03h Unified cache
-                // 1Fh-04h Reserved.
+                // 000b Reserved.
+                // 001b Level 1
+                // 010b Level 2
+                // 011b Level 3
+                // 111b-100b Reserved.
                 // ```
                 //
-                // cache_type: 0..4,
-                let cache_level = 15 & subleaf.result.eax; // 15 == (2^4)-1
+                // cache_level: 5..8
+                let cache_level = get_range(&subleaf.result.eax, 5..8);
```

Comment:
I'm not convinced it's correct for AMD. The indexes I can see in the spec (search for `CPUID_Fn8000001D_EAX_x00) are:
- DC (L1)
- IC (L1)
- L2
- L3
- Null

Reply:
From the spec:

![image](https://github.com/firecracker-microvm/firecracker/assets/16856679/642e89d3-3f75-4d71-9992-c2deabc2d6fe)

Reply:
That doesn't explain it though, does it?
How would we interpret `Cache Properties (DC)` using the attached page 628?

Reply:
I think we should be iterating over the subleaves 0..4 (4 being Null) and setting the number of processors sharing the cache based on the cache level value in the subleaf itself instead of relying on the iterator variable.

Reply:
```rust
for subleaf in 0..4 {
  let cache_level = ...;
  match cache_level {
     1 | 2 => <set 1 or 2>;
     3 => <set cpu_count>;
  }
}
```

Reply:
I believe there are often more than 4 subleaves.

Reply:
Yes, we discussed this offline. This should be the exit criterion, but currently we do not do this here or in main.

Reply:
indeed, you're right!